source/common/dynamo/dynamo_filter.cc

```diff
@@ -32,6 +32,16 @@ Http::FilterDataStatus DynamoFilter::decodeData(Buffer::Instance& data, bool end
     onDecodeComplete(data);
   }
 
+  // If the request is too large, send a 413 to the user.   This could be applied before the check
+  // above, but as Envoy buffer limits are soft limits, if we already have the data buffered might
+  // as well pass it through.
+  if (buffer_limit_ > 0 && data.length() > buffer_limit_) {
```

Comment:
This filter pushes buffering down to the connection manager. In this case, should we rely on the connection manager to actually send the 413 and reset? Very few filters will end up doing internal buffering that is outside of what the connection manager will provide IMO. Just something to think about.